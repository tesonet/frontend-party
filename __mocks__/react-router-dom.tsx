import React, { ReactChildren } from "react";
import * as rrd from "react-router-dom";

const MockBrowserRouter: React.FC<{ children: ReactChildren }> = ({
  children
}) => <div>{children}</div>;

/* eslint-disable @typescript-eslint/no-explicit-any */
(rrd as any).BrowserRouter = MockBrowserRouter;

module.exports = rrd;
