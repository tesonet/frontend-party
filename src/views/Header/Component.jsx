import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import styles from './styles.scss';

const Header = props => (
  <div className={cn('row', styles.header)}>
    <div className={cn('col-xs-8', styles.main)}>
      <img
        src="/static/img/logo.png"
        alt="Testio"
        width={115}
        height={30}
        className={styles.logo}
      />
    </div>
    <div className={cn('col-xs-4', styles.aside)}>
      <button onClick={props.logout}>Logout</button>
    </div>
  </div>
);

Header.propTypes = {
  logout: PropTypes.func.isRequired,
};

export default Header;
