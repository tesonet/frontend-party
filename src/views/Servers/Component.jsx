import React from 'react';
import Icon from 'react-fa';
import PropTypes from 'prop-types';
import cn from 'classnames';
import ServerRow from '../../components/ServerRow';

import styles from './styles.scss';

class Servers extends React.Component {
  componentDidMount() {
    this.props.getServers();
  }

  sortItems = (a, b) => (a.distance - b.distance) || a.name.localeCompare(b.name);

  renderItems = () => (
    <div>
      <ServerRow item={{ name: 'SERVER' }} isHeader />
      {this.props.items.sort(this.sortItems).map(this.renderItem)}
    </div>
  );

  renderLoading = () => <div className={styles.loading}><Icon spin name="spinner" /></div>;

  renderItem = (item, index) => <ServerRow key={index} item={item} />;

  render() {
    return (
      <div className={cn(styles.container)}>
        {this.props.loading && this.renderLoading()}
        {!!this.props.items.length && this.renderItems()}
      </div>
    );
  }
}

Servers.propTypes = {
  getServers: PropTypes.func.isRequired,
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  loading: PropTypes.bool,
};

Servers.defaultProps = {
  loading: false,
};

export default Servers;
