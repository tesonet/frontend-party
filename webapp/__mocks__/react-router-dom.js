import React from 'react';

const reactRouterDom = require('react-router-dom');

// Mocks the BrowserRouter component so that it doesn't collide with the MemoryRouter
// that's used in tests.
reactRouterDom.BrowserRouter = ({ children }) => children;

module.exports = reactRouterDom;
