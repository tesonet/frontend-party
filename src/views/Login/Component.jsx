import React from 'react';
import Icon from 'react-fa';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import cn from 'classnames';
import styles from './styles.scss';

class Login extends React.Component {
  handleSubmit = (e) => {
    e.preventDefault();
    const username = e.target.elements.username.value;
    const password = e.target.elements.password.value;
    if (username && password) {
      this.props.login(username, password);
    }
  };

  render() {
    const { from } = this.props.location.state || { from: { pathname: '/' } };

    if (this.props.token) {
      return <Redirect to={from} />;
    }

    return (
      <div className={cn('container', styles.container)}>
        <form className={styles.form} onSubmit={this.handleSubmit}>
          <img
            src="/static/img/logo-front.png"
            alt="Testio"
            width={246}
            height={65}
            className={styles.logo}
          />
          <p><input type="text" name="username" required placeholder="Username" /></p>
          <p><input type="password" name="password" required placeholder="Password" /></p>
          {!!this.props.error && <p className={styles.error}>{this.props.error}</p>}
          <p>
            <button type="submit" disabled={this.props.loading}>
              {this.props.loading ? <Icon spin name="spinner" /> : 'Log In'}
            </button>
          </p>
        </form>
      </div>
    );
  }
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
  token: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]).isRequired,
  location: PropTypes.PropTypes.shape({
    state: PropTypes.string,
  }).isRequired,
  error: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]),
  loading: PropTypes.bool.isRequired,
};

Login.defaultProps = {
  error: '',
};

export default Login;
