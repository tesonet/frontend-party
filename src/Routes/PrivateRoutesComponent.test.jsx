import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Component from './PrivateRoutesComponent';

test('Dont show private content when no token is found', () => {
  const component = renderer.create(
    <StaticRouter context={{}}>
      <Component token={false} location={{ pathname: '/' }} />
    </StaticRouter>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Show private content when no token is found', () => {
  const store = {
    subscribe: () => {},
    dispatch: () => {},
    getState: () => ({
      login: {
        token: 'token',
      },
      servers: {
        items: [],
      },
    }),
  };
  const component = renderer.create(
    <Provider store={store}>
      <StaticRouter context={{}}>
        <Component token={'token'} location={{ pathname: '/' }} />
      </StaticRouter>
    </Provider>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
