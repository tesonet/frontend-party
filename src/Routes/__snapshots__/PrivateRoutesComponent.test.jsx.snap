// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dont show private content when no token is found 1`] = `null`;

exports[`Show private content when no token is found 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row header"
  >
    <div
      className="col-xs-8 main"
    >
      <img
        alt="Testio"
        className="logo"
        height={30}
        src="/static/img/logo.png"
        width={115}
      />
    </div>
    <div
      className="col-xs-4 aside"
    >
      <button
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
  <div
    className="container"
  />
</div>
`;
